ci/fix_wheels.py

```diff
@@ -0,0 +1,55 @@
+import os
+import shutil
+import struct
+import sys
+import zipfile
+
+try:
+    _, wheel_path, dest_dir = sys.argv
+    # Figure out whether we are building on 32 or 64 bit python
+    is_32 = struct.calcsize("P") * 8 == 32
+    PYTHON_ARCH = "x86" if is_32 else "x64"
+except ValueError:
+    # Too many/little values to unpack
+    raise ValueError(
+        "User must pass the path to the wheel and the destination directory."
+    )
+# Wheels are zip files
+if not os.path.isdir(dest_dir):
+    print(f"Created directory {dest_dir}")
+    os.mkdir(dest_dir)
+shutil.copy(wheel_path, dest_dir)  # Remember to delete if process fails
+wheel_name = os.path.basename(wheel_path)
+success = True
+exception = None
+repaired_wheel_path = os.path.join(dest_dir, wheel_name)
+with zipfile.ZipFile(repaired_wheel_path, "a") as zipf:
```

Comment:
Is there a reason for using zipfile versus wheel? Not sure what implementation differences there are but likely safer to use `wheel unpack`

Reply:
Wheels are zip files underneath. I don't think using wheel is a good idea since wheel has no public Python API, and I don't want to do a bunch of subprocess calls here.

Reply:
Yea that's a bit unfortunate with wheel. The other consideration point is that the zipfile is mostly an implementation detail of wheel at the moment; may or may not last forever. There's been some talk about this over the past few years

https://discuss.python.org/t/making-the-wheel-format-more-flexible-for-better-compression-speed/3810

There's also apparently some file hashes that wheel takes care of that you wouldn't get this way (saw in wheel documentation). Not sure of the ultimate impacts of that

Reply:
Some interesting discussion here as well

https://github.com/pypa/wheel/issues/262

Reply:
I think we can stick with this for now. I'm just not a huge fan of The file hashes might be important, but given that pip installs the wheel without complaint, I'm going to ignore it for now. If anything fails, it will fail loudly in the future.